projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-posts.php

```diff
@@ -27,6 +27,23 @@ public function setUp() {
 		$this->sender->do_sync();
 	}
 
+	/**
+	 * Verify post_content is limited based on MAX_POST_CONTENT_LENGTH.
+	 */
+	public function test_post_content_limit() {
+
+		$post_sync_module = Modules::get_module( 'posts' );
+
+		$this->post->post_content = str_pad( '', Automattic\Jetpack\Sync\Modules\Posts::MAX_POST_CONTENT_LENGTH - 1, 'X', STR_PAD_LEFT );
+		$filtered_post            = $post_sync_module->filter_post_content_and_add_links( $this->post );
+		$this->assertNotEmpty( $filtered_post->post_content, 'Filtered post content is empty for stings of allowed length.' );
+
+		$this->post->post_content = str_pad( '', Automattic\Jetpack\Sync\Modules\Posts::MAX_POST_CONTENT_LENGTH + 1, 'X', STR_PAD_LEFT );
```

Comment:
Same.

Also, you test MAX-1 and MAX+1, but no test for exactly MAX. As your code is currently written, MAX gets the content removed too, so
```suggestion
		$this->post->post_content = str_repeat( 'X', Automattic\Jetpack\Sync\Modules\Posts::MAX_POST_CONTENT_LENGTH );
```